### DISCUSSION TITLE
Drop support for Sigv2

### DISCUSSION THREAD

**Discussion by FlorianSW** [nginxinc/nginx-s3-gateway#46] [Ideas]
Even though, existing buckets are still able to use Sigv2 for S3 (new buckets can't do so for quite some time now already), I don't see any legitimate argument, why this gateway should support Sigv2 any longer:
https://aws.amazon.com/blogs/aws/amazon-s3-update-sigv2-deprecation-period-extended-modified/

- every bucket is able to use Sigv4 (as far as I know)
- Only a small subset of regions even still support Sigv2 (https://docs.aws.amazon.com/general/latest/gr/signature-version-2.html)
- Implementing new features (like #8) might get harder to implement when both signature versions are going to be supported (as both versions need to be maintained)

What do others think about that? :)

**Comment by dekobon** [comment]
I see the following arguments for keeping support:

1. This project isn't really a software product, but just a ready to use example of how to configure NGINX for using signatures for authentication. As such, dropping an example may result in losing a useful reference.
2. Some S3 compatible services exclusively rely on v2 signatures.
3. Some S3 compatible services prefer v2 signatures for performance reasons. Wasabi comes to mind. As I understand it, they guide their customers to use v2 signature despite supporting v4 signatures because they are generally more performant.

Probably #2 is the biggest argument to keep it. I would need to do a survey of S3 compatible services/shims out there to see where v4 support is at.

**Comment by dekobon** [comment]
We could also easily enough just break the v2 supporting gateway into another directory and build a different container image for it.
